tsdb/engine/tsm1/compact.go

```diff
@@ -1061,16 +1073,15 @@ func (c *Compactor) writeNewFiles(generation, sequence int, src []string, iter K
 			// planner keeps track of which files are assigned to compaction plans now.
 			return nil, err
 		} else if err != nil {
+			// We hit an error and didn't finish the compaction.  Abort.
 			// Remove any tmp files we already completed
+			// discard later errors to return the first one from the write() call
 			for _, f := range files {
-				if err := os.RemoveAll(f); err != nil {
```

Comment:
Why the change in error semantics, and does this risk breaking error handling expectations elsewhere, e.g. in compactionStrategy.compactGroup()? I don't have enough clarity on what kinds of errors, in what order, would be returned in the unroll by the old code.

Reply:
We hit an error from the c.write() call. That's the most important error to log, because the first error has the most information about the initial problem. Overwriting it with a subsequent error from a clean-up call can mask the root of the problem. We don't care what errors the RemoveAll() calls return, because we failed before them.